### PR TITLE
Enhancement: Boxes Indexer Integration test

### PR DESCRIPTION
**Pull request by ahangsu**

### CODE REVIEW

File: .test-env [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Configs for testing repo download:
 SDK_TESTING_URL="https://github.com/algorand/algorand-sdk-testing"
-SDK_TESTING_BRANCH="feature/box-storage"
+SDK_TESTING_BRANCH="ahangsu/indexer-integration-boxes"
```
Comment (ahangsu) [inline .test-env:3]: TODO: when `feature/box-storage` PR is merged, do
```suggestion
SDK_TESTING_BRANCH="feature/box-storage"
```

File: test/applications_integration_test.go [hunk 2]
```diff
@@ -874,52 +1010,9 @@ func ApplicationsContext(s *godog.Suite) {
 	s.Step(`^The (\d+)th atomic result for randomInt\((\d+)\) proves correct$`, checkRandomIntResult)
 	s.Step(`^The (\d+)th atomic result for randElement\("([^"]*)"\) proves correct$`, checkRandomElementResult)
 
-	s.Step(`^the contents of the box with name "([^"]*)" in the current application should be "([^"]*)"\. If there is an error it is "([^"]*)"\.$`, theContentsOfTheBoxWithNameShouldBeIfThereIsAnErrorItIs)
-
-	s.Step(`^the current application should have the following boxes "([^"]*)"\.$`,
```
Comment (tzaffi) [inline test/applications_integration_test.go:879]: nice refactor!

File: test/applications_integration_test.go [hunk 3]
```diff
@@ -846,6 +856,132 @@ func theContentsOfTheBoxWithNameShouldBeIfThereIsAnErrorItIs(encodedBoxName, box
...
+		return err
+	}
+
+	if len(r.Boxes) != expectedNum {
+		return fmt.Errorf("expected and actual box number do not match: %v != %v", expectedNum, len(r.Boxes))
+	}
+	return nil
+}
+
+func indexerSaysCurrentAppShouldHaveTheseBoxes(max int, next string, encodedBoxesRaw string) error {
```
Comment (tzaffi) [inline test/applications_integration_test.go:940]: Is this step still needed? I see that `currentApplicationShouldHaveFollowingBoxes` has logic for indexer as well as algod.

**Review by tzaffi — COMMENTED** [review]
same tag issue + possibly obsolete function?

File: test/applications_integration_test.go (hunk 3 above)
Comment (tzaffi) [reply test/applications_integration_test.go:940]: Yes it is because of the `next` parameter.

Resolving

**Review by tzaffi — APPROVED** [review]
LGTM

**Comment by tzaffi** [comment]
> LGTM

LEBTM after most recent changes

File: test/applications_integration_test.go [hunk 4]
```diff
@@ -828,9 +831,16 @@ func decodeBoxName(encodedBoxName string) ([]byte, error) {
...
-	box, err := algodV2client.GetApplicationBoxByName(applicationId).Name(encodedBoxName).Do(context.Background())
+func theContentsOfTheBoxWithNameShouldBeIfThereIsAnErrorItIs(fromClient, encodedBoxName, boxContents, errStr string) error {
+	var box models.Box
+	var err error
+	if fromClient == "algod" {
+		box, err = algodV2client.GetApplicationBoxByName(applicationId).Name(encodedBoxName).Do(context.Background())
+	} else if fromClient == "indexer" {
+		box, err = indexerV2client.LookupApplicationBoxByIDandName(applicationId).Name(encodedBoxName).Do(context.Background())
+	} else {
+		err = fmt.Errorf("expecting algod or indexer, get " + fromClient)
```
Comment (algochoi) [inline test/applications_integration_test.go:842]:
```suggestion
		err = fmt.Errorf("expecting algod or indexer, got " + fromClient)
```

**Review by algochoi — APPROVED** [review]
Looks good!

**Pull request merged**